src/main/java/com/yelp/nrtsearch/server/luceneserver/IndexState.java

```diff
@@ -641,6 +646,9 @@ public synchronized void load(JsonObject jsonObject) throws IOException {
   @Override
   public void close() throws IOException {
     logger.info(String.format("IndexState.close name= %s", name));
+    if (isPeriodicCommitEnabled()) {
+      periodicCommit.cancel();
+    }
```

Comment:
Might be better to just make the class extend closable and add it to the closables list.

Reply:
Might be tricky since we want to stop the thread before the shard is closed (to not call commit after the shard is closed).

Reply:
The closables list is processed in order, so this could be done first. But there should be a comment explaining the ordering dependency.